classes/Pods.php

```diff
@@ -1807,10 +1810,26 @@ public function field( $name, $single = null, $raw = false ) {
 												$metadata_type = 'term';
 											}
 
-											$value[] = get_metadata( $metadata_type, $metadata_object_id, $field, true );
+											$maybe_traverse_val = get_metadata( $metadata_type, $metadata_object_id, $field, true );
+											$maybe_traverse     = true;
 										} elseif ( 'settings' === $object_type ) {
-											$value[] = get_option( $object . '_' . $field );
+											$maybe_traverse_val = get_option( $object . '_' . $field );
+											$maybe_traverse     = true;
 										}//end if
+
+										if ( isset( $maybe_traverse_val ) ) {
+											if ( $maybe_traverse && is_array( $maybe_traverse_val ) && 1 < count( $traverse_fields ) ) {
+												array_shift( $traverse_fields );
+												foreach ( $traverse_fields as $field ) {
+													if ( ! isset( $maybe_traverse_val[ $field ] ) ) {
+														break;
```

Comment:
@JoryHogeveen If we find that the traverse field is not set in the value, should we return a null or empty string for the `$maybe_traverse_val`?

```suggestion
														$maybe_traverse_val = null;

														break;
```